doc/development/deprecations.rst

```diff
@@ -9,6 +9,26 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* The private functions ``_pauli_mult``, ``_binary_matrix`` and ``_get_pauli_map`` from the
+  ``pauli`` module have been deprecated, as they are no longer used anywhere and the same
+  functionality can be achieved using newer features in the ``pauli`` module.
+
+  - Deprecated in v0.35
+  - Will be removed in v0.36
+
+* ``qml.pauli.pauli_mult`` and ``qml.pauli.pauli_mult_with_phase`` are now deprecated. Instead, you
+  should use ``qml.prod(pauli_1, pauli_2).simplify()`` to get the reduced operator. Note that if there
+  is a phase, you can access it via ``op.scalar``.
+
+  >>> op = qml.prod(qml.PauliX(0), qml.PauliZ(0)).simplify()
+  >>> op, op.scalar, op.base
+  (-1j*(PauliY(wires=[0])), -1j, PauliY(wires=[0]))
+  >>> qml.prod(qml.PauliZ(0) @ qml.PauliX(0)).simplify()
+  PauliZ(wires=[0]) @ PauliX(wires=[0])
```

Comment:
What's the point of this part of the example? Also, with `simplify()` shouldn't we get a `PauliY`?

Reply:
omg 🤦 It was to show that sometimes, the result isn't an SProd and doesn't have `scalar` and `base` properties, but this _should_ be an SProd (like you said, `1j * Y0`). It didn't happen because I used `@` instead of a comma when constructing the op. will fix